File: 3242-count-elements-with-maximum-frequency/count-elements-with-maximum-frequency.cpp
class Solution {
public:
    int maxFrequencyElements(vector<int>& nums) {
        int n = nums.size();
        unordered_map<int, int> mp;
        int maxi = 0;
        int ans = 0;

        for(int i=0; i<n; i++){
            mp[nums[i]]++;
            if(maxi < mp[nums[i]]){
                maxi = mp[nums[i]];
            }
        }

        for(auto it : mp){
            if(it.second == maxi){
                ans += it.second;
            }
        }

        return ans;
    }
};

File: 3242-count-elements-with-maximum-frequency/count-elements-with-maximum-frequency.java
class Solution {
    public int maxFrequencyElements(int[] nums) {
        Map<Integer, Integer> mp = new HashMap<>();
        int maxFreq = 0;
        int ans = 0;

        for(int i=0; i<nums.length; i++){
            mp.put(nums[i], mp.getOrDefault(nums[i], 0) + 1);
            if(maxFreq < mp.get(nums[i])){
                maxFreq = mp.get(nums[i]);
            }
        }

        for(int num : mp.keySet()){
            if(mp.get(num) == maxFreq) ans += maxFreq;
        }

        return ans;
    }
}